### DISCUSSION TITLE
Linux firmware updater added in boot menu

### DISCUSSION THREAD

**Discussion by berlinwn** [fwupd/fwupd#6014] [General]
When updating a firmware in Linux,the Linux firmware updater option is added to the boot menu.

Can you tell why the Linux firmware updater was added?

**Comment by hughsie** [comment]
Sure! We use a EFI loader to stage the capsule -- we need to use an EFI method and it's easier to do from a small binary run before grub or shim. See https://github.com/fwupd/fwupd/blob/main/plugins/uefi-capsule/README.md and https://github.com/fwupd/fwupd-efi for more details.

**Reply by berlinwn** [reply]
Can you tell how do disable the "Linux firmware updater " in the boot menu

**Reply by hughsie** [reply]
efibootmgr should be able to delete it -- although it's harmless

**Reply by berlinwn** [reply]
Hi hughsie,
"Linux firmware updater" is added in the boot menu because of a boot manager bug. Say your thoughts?

**Reply by hughsie** [reply]
It's not a bug, we leave it there intentionally to **avoid** triggering firmware bugs when removing entries.

**Reply by berlinwn** [reply]
Can "Linux firmware updater" be remove automatically?

**Reply by hughsie** [reply]
Well, it could, but why?

**Reply by berlinwn** [reply]
After updating the firmware in Linux, The Linux firmware updater is added to the bootmenu.It cause confusion in boot menu.
Can you tell the sollution to remove the Linux firmware updater Automatically?

**Reply by superm1** [reply]
Does removing it from BootOrder hide from the menu?

**Reply by berlinwn** [reply]
Yes,it removed but after repeat the firmware update,The "Linux firmware updater" again added in the Boot menu.

Can you tell the "Linux firmware updater" is removed Automatically?

**Reply by superm1** [reply]
Right; so my suggestion was going to be that we keep the BootXXXX entry, but we just drop it from BootOrder on the next boot.

**Reply by hughsie** [reply]
Maybe we drop the "add to BootOrder" compat hack? we have

	/* TODO: conditionalize this on the UEFI version? */
	if (!fu_uefi_bootmgr_add_to_boot_order(boot_next, error))
		return FALSE;

and I don't think we need that block at all. Maybe we add it to a uefi_capsule.conf (default off) as a config value and then rip it out in 1y if nobody complains.

**Reply by superm1** [reply]
Sure that sounds good

**Reply by hughsie** [reply]
https://github.com/fwupd/fwupd/pull/6070
